darwin/datetimepicker.m

```diff
@@ -4,17 +4,90 @@
 struct uiDateTimePicker {
 	uiDarwinControl c;
 	NSDatePicker *dp;
+	void (*onChanged)(uiDateTimePicker *, void *);
+	void *onChangedData;
 };
 
+@interface datePickerDelegateClass : NSObject <NSDatePickerCellDelegate> {
+	struct mapTable *pickers;
+}
+- (void)datePickerCell:(NSDatePickerCell *)aDatePickerCell
+	validateProposedDateValue:(NSDate **)proposedDateValue
+	timeInterval:(NSTimeInterval *)proposedTimeInterval;
+- (void)doTimer:(NSTimer *)timer;
+- (void)registerPicker:(uiDateTimePicker *)b;
+- (void)unregisterPicker:(uiDateTimePicker *)b;
+@end
+
+@implementation datePickerDelegateClass
+
+- (id)init
+{
+	self = [super init];
+	if (self)
+		self->pickers = newMap();
+	return self;
+}
+
+- (void)dealloc
+{
+	mapDestroy(self->pickers);
+	[super dealloc];
+}
+
+- (void)datePickerCell:(NSDatePickerCell *)aDatePickerCell
```

Comment:
NSDatePicker is a NSControl, so target-action *should* work and you won't need to fiddle with `NSTimer` or delegates. If it doesn't, could you add a comment explaining that (and consider using `dispatch_async()` instead of `NSTimer`)?

Reply:
`- (void)datePickerCell:` allows you to modify or even reject the event via `*proposedDateValue`, so `dateValue` isn't valid until the method returns. The timer moves the actual event firing to the next run of the event loop, and it could totally use `dispatch_async()` instead of `NSTimer` if it makes a difference.

Reply:
`- (void)datePickerCell:` allows you to modify or even reject the event via `*proposedDateValue`, so `dateValue` isn't valid until the method returns. The timer moves the actual event firing to the next run of the event loop, and it could totally use `dispatch_async()` instead of `NSTimer` if it makes a difference.

Reply:
Using `dispatch_async()` looks more correct, and comes with the guarantee of being run when next possible, rather than being subject to any delays introduced by NSTimer.